src/Ship.js

```diff
@@ -1,8 +1,24 @@
-function Ship(startingPort) {
-    this.startingPort = startingPort;
-    this.setSail = function () {
-        this.startingPort = null;
+class Ship {
+    constructor(itinerary) {
+        this.itinerary = itinerary;
+        this.currentPort = itinerary.ports[0];
+        this.previousPort = null;
+        this.currentPort.addShip(this);
     }
+        setSail() {
+            this.previousPort = this.currentPort;
+            this.currentPort = null;
+            this.previousPort.removeShip(this)
+        }
+        dock() {
+            const itinerary = this.itinerary;
```

Comment:
No need to re-assign here.
this.itinerary exists in your constructor anyway, so you can use it directly in the next line.